clippy_lints/src/new_without_default.rs

```diff
@@ -118,20 +118,26 @@ impl LateLintPass for NewWithoutDefault {
                                                      `Default` implementation for `{}`",
                                                     self_ty),
                                            |db| {
-                                               db.span_suggestion(span, "try this", "#[derive(Default)]".into());
-                                           });
+                            db.suggest_item_with_attr(cx, span, "try this", "#[derive(Default)]");
+                        });
                     } else {
                         span_lint_and_then(cx,
                                            NEW_WITHOUT_DEFAULT, span,
                                            &format!("you should consider adding a \
                                                     `Default` implementation for `{}`",
                                                     self_ty),
                                            |db| {
-                                               db.span_suggestion(span,
-                                                                  "try this",
-                                                                  format!("impl Default for {} {{ fn default() -> \
-                                                                          Self {{ {}::new() }} }}", self_ty, self_ty));
-                                           });
+                        db.suggest_prepend_item(cx,
+                                                  span,
+                                                  "try this",
+                                                  &format!(
+"impl Default for {} {{
```

Comment:
If you indent the string, all of the indentation is included except if you use `\` and then none of the indentation is included:

```rust
    "
    foo
    bar
        baz" == "\n    foo\n    bar\n        baz";

"\
foo\
    bar\
        baz" == "foobarbaz"
```